src/DiIiS-NA/D3-GameServer/GSSystem/AISystem/Brains/MinionBrain.cs

```diff
@@ -196,13 +197,12 @@ protected virtual int PickPowerToUse()
 			// randomly used an implemented power
 			if (PresetPowers.Count > 0)
 			{
-				//int power = this.PresetPowers[RandomHelper.Next(this.PresetPowers.Count)].Key;
-				List<int> availablePowers = Enumerable.ToList(PresetPowers.Where(p => (p.Value.CooldownTimer == null || p.Value.CooldownTimer.TimedOut) && PowerLoader.HasImplementationForPowerSNO(p.Key)).Select(p => p.Key));
-				if (availablePowers.Where(p => p != 30592).Count() > 0)
```

Comment:
1 filter to check count + 1 filter to make list + 1 filter to get random index.
Now we can filter once by using TryPickRandom